de-lib/src/main/java/org/iplantc/de/collaborators/client/presenter/ManageCollaboratorsPresenter.java

```diff
@@ -112,14 +113,49 @@ public void onUserSearchResultSelected(UserSearchResultSelected userSearchResult
         Subject subject = userSearchResultSelected.getSubject();
         if (!userInfo.getUsername()
                      .equals(subject.getId())) {
-            if (!collaboratorsUtil.isCurrentCollaborator(subject, view.getCollaborators())) {
-                addAsCollaborators(Arrays.asList(subject));
+            if (view.hasCollaboratorsTabSelected()) {
+                if (!collaboratorsUtil.isCurrentCollaborator(subject, view.getCollaborators())) {
+                    addAsCollaborators(Arrays.asList(subject));
+                }
+            } else {
+                List<Group> selectedCollaboratorLists = view.getSelectedCollaboratorLists();
+                if (selectedCollaboratorLists != null && !selectedCollaboratorLists.isEmpty()) {
+                    addMemberToGroups(subject, selectedCollaboratorLists);
+                } else {
+                    announcer.schedule(new ErrorAnnouncementConfig(groupAppearance.noCollabListSelected()));
+                }
             }
         } else {
             announcer.schedule(new ErrorAnnouncementConfig(I18N.DISPLAY.collaboratorSelfAdd()));
         }
     }
 
+    void addMemberToGroups(Subject subject, List<Group> selectedCollaboratorLists) {
+        selectedCollaboratorLists.forEach(new Consumer<Group>() {
+            @Override
+            public void accept(Group group) {
+                groupServiceFacade.addMembers(group, Lists.newArrayList(subject), new AsyncCallback<List<UpdateMemberResult>>() {
```

Comment:
Is there way to avoid `RPC` calls from a loop? or Somehow combine the results of all the calls and finally display one error message ? Throwing multiple errors may lead to unpleasant user experience.